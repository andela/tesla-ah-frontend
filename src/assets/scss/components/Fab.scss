.fab {
  display: none;
}

@media (max-width: 991px) {
  .fab {
    display: block;
  }
}