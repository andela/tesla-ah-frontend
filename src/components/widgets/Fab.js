import React from 'react';
import { Fab as FabButton, Action } from 'react-tiny-fab';

import '../../assets/scss/components/Fab.scss';
import '../../../node_modules/react-tiny-fab/dist/styles.css';

const Fab = (props) => {
  const {
    articles,
    followers,
    following,
    isCurrentUser,
  } = props;
  const actionStyles = { backgroundColor: '#00aeff', color: '#fff' };
  return (
    <div className="fab">
      <FabButton
        mainButtonStyles={{ ...actionStyles, backgroundColor: '#303346' }}
        position={{ bottom: 24, right: 24 }}
        icon={<i className="fas fa-bars" />}
        event="hover"
      >
        <Action id="fabArticles" text="Articles" style={actionStyles} onClick={articles}>
          <i className="fas fa-feather" />
        </Action>
        <Action text="Followers" style={actionStyles} onClick={followers}>
          <i className="fas fa-users" />
        </Action>
        <Action text="Following" style={actionStyles} onClick={following}>
          <i className="fas fa-forward" />
        </Action>
        {isCurrentUser ? (
          <Action
            text="Edit Profile"
            style={actionStyles}
            data-toggle="modal"
            data-target="#exampleModalCenter"
          >
            <i className="fas fa-user-edit" />
          </Action>
        ) : null }
      </FabButton>
    </div>
  );
};

export default Fab;
