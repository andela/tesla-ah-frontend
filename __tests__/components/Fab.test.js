/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Fab from '../../src/components/widgets/Fab';
import { profile } from '../mockData';

const renderFab = (args) => {
  const defaultProps = {
    articles: jest.fn(),
    followers: jest.fn(),
    following: jest.fn(),
  };
  const props = { ...defaultProps, ...args };

  return shallow(<Fab {...props} />);
};

it('renders Fab with all actions', () => {
  const wrapper = renderFab();
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('Fab').length).toBe(1);
});

it('renders Fab with edit', () => {
  const wrapper = renderFab({ isCurrentUser: profile.currentUser });
  expect(wrapper.find('Fab').length).toBe(1);
});
