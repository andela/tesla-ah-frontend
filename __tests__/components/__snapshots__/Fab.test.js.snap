// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Fab with all actions 1`] = `ShallowWrapper {}`;
